canonical-web-and-design/maas-ui
# refactor(ui): handle undefined URL params

## Done

- Add hooks and utils for getting and parsing ids from the URL.
- Replace useParams calls with the id hook.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- View some details pages (e.g. machine details, devices details etc.) and check that they load.

## Fixes

Fixes: #3456.
